### PR TITLE
MacOS OpenVPN and DNS Manager

### PR DESCRIPTION
**Pull request by glowinthedark**
Bitbar plugin for controlling OpenVPN connection profiles and setting DNS configurations.

Preconditions
----------

 - INSTALL OPENVPN:

```bash
brew install openvpn
```
> NOTE: To avoid having to enter the root password allow passwordless sudo for your user on: `openvpn`, `killall` and `pkill`:

1. Type in terminal: `sudo visudo`
2. Append the following line at the end of the file:

```
    your_username   ALL = NOPASSWD: /usr/local/sbin/openvpn, /usr/bin/killall, /usr/bin/pkill
```

3. Search and replace all `__FIXME__` lines with your specific commands (for DNS and ping servers, passwords, OTP tokens)
4. [ON FIRST RUN ONLY] Run the script and grant MacOS accessibility permissions to Terminal.app, bitbar/xbar (whichever applicable), and to `/usr/bin/osascript`.

### CODE REVIEW

**Comment by glowinthedark** [comment]
@matryer: what is the proper way to programmatically refresh the plugin via a command in the plugin with xbar? 

I am using the old (bitbar) syntax:

```bash
open -g "bitbar://refreshPlugin?name=openvpn.*?.sh"
``` 

but the command above launches the old BitBar.app rather than refreshing the xbar plugin.

I tried changing it to:
```bash
open -g "xbar://refreshPlugin?name=openvpn.*?.sh"
``` 
but this throws an `Invalid URL` error.

File: Network/openvpn.5m.sh [hunk 1]
```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env bash
+# <bitbar.title>MacOS OpenVPN client & DNS Configuration</bitbar.title>
+# <bitbar.version>v0.88008</bitbar.version>
+# <bitbar.author>glowinthedark</bitbar.author>
+# <bitbar.author.github>glowinthedark</bitbar.author.github>
+# <bitbar.desc>OpenVPN GUI & DNS Manager</bitbar.desc>
+# <bitbar.image>https://telegra.ph/file/7e93fd31b281c9cdcab73.png</bitbar.image>
+# <bitbar.dependencies>openvpn</bitbar.dependencies>
+# <bitbar.abouturl>https://github.com/glowinthedark/bitbar-plugins/blob/master/Network/openvpn.5m.sh</bitbar.abouturl>
```
Comment (sprak3000) [inline Network/openvpn.5m.sh:9]:
```suggestion
# <xbar.title>MacOS OpenVPN client & DNS Configuration</xbar.title>
# <xbar.version>v0.88008</xbar.version>
# <xbar.author>glowinthedark</xbar.author>
# <xbar.author.github>glowinthedark</xbar.author.github>
# <xbar.desc>OpenVPN GUI & DNS Manager</xbar.desc>
# <xbar.image>https://telegra.ph/file/7e93fd31b281c9cdcab73.png</xbar.image>
# <xbar.dependencies>openvpn</xbar.dependencies>
# <xbar.abouturl>https://github.com/glowinthedark/bitbar-plugins/blob/master/Network/openvpn.5m.sh</xbar.abouturl>
```

File: Network/openvpn.5m.sh [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+# Under System Preferences select:
+#     > Security & Privacy
+#     > Privacy tab 
+#     > select [Accessibily] in the left panel 
+#     > click the 🔒 icon to unlock and add Bitbar.app, Terminal.app and /usr/bin/osascript
+#     (if the apps are already enabled in the list, then remove them and re-add)
+# In System Settings: allow Terminal full access to Documents (if profiles are in Documents) or full disk access
+
+# START CONFIG
+export PATH="$PATH:/usr/local/bin"
```
Comment (sprak3000) [inline Network/openvpn.5m.sh:47]: M1 based Macs install Homebrew to a new location. Not sure if the line below needs to account for this.

```suggestion
export PATH="$PATH:/opt/homebrew/bin:/usr/local/bin"
```

File: Network/openvpn.5m.sh [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+# NOTE: To avoid having to enter the root password allow
+# passwordless sudo for your user on openvpn, killall and pkill:
+
+# 1. Type in terminal: sudo visudo
+# 2. Append the following line at the end of the file:
+
+# replace_with_your_username        ALL = NOPASSWD: /usr/local/sbin/openvpn, /usr/bin/killall, /usr/bin/pkill
+
+# 3. Search and replace all __TODO__ lines with your specific commands
```
Comment (sprak3000) [inline Network/openvpn.5m.sh:27]: Would this be a good use case for [xbar variables](https://xbarapp.com/docs/2021/03/14/variables-in-xbar.html)?

Comment (glowinthedark) [reply Network/openvpn.5m.sh:27]: @sprak3000: oh! this is something new I was not aware of; in fact, I'm still using the latest tip of the old and discontinued objective-c version; the new go+webview based version uses more CPU on my system, and also with xbar I didn't yet find an equivalent of the previous programmatic refresh with `open -g "bitbar://refreshPlugin?name=openvpn.*?.sh"`; any suggestions on this?

there would probably be quite a few variables to pull into the config dialog; a good challenge for a next iteration :D

**Review by sprak3000 — APPROVED** [review]

File: Network/openvpn.5m.sh (hunk 3 above)
Comment (sprak3000) [reply Network/openvpn.5m.sh:27]: @glowinthedark 

There is a new `refresh` [parameter](﻿﻿https://github.com/matryer/xbar-plugins/blob/main/CONTRIBUTING.md#parameters). Haven't used it myself or the previous way you mention, so I do not know if they are equivalent.

Thank you for your contribution!

**Pull request merged**